### PR TITLE
Signup with side effects

### PR DESCRIPTION
**Pull request by enricostano**
#### Create
Request:
```
POST api/v1/signups

{
  "email": "user@email.com"
}
```
Response:
```
201 CREATED

{}
```
#### Show
Request:
```
GET api/v1/signups/:token
```
Response:
```
200 OK

{
  "email": "user@email.com"
}
```
#### Complete
Request:
```
POST api/v1/signups/complete/:token

{
  "username": "username",
  "password": "sosecret",
  "password_confirmation": "sosecret",
  "first_name": "User",
  "last_name": "Name"
}
```
Response:
```
200 OK

{
  "id": 1,
  "email": "user@email.com",
  "username": "username",
  "first_name": "User",
  "last_name": "Name"
  "created_at": "2014-09-06T11:48:39.072Z",
  "updated_at": "2014-09-06T11:48:39.072Z"
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,22 +4,62 @@ Ruby on Rails app to foster collaborative consumption
...
+200 OK
+
+{
+  "email": "user@email.com"
+}
+```
+#### Complete
+Request:
+```
+POST api/v1/signups/:token
```
Comment (andresgutgon) [inline README.md]: `api/v1/signups/complete/:token`

File: engines/account/app/controllers/account/api/v1/signups_controller.rb [hunk 2]
```diff
@@ -23,24 +25,61 @@ def create
         # TODO: better serialized response
         #
         def show
-          signup = ::Account::Signup.find_by_token(signup_show_params[:token])
+          render status: :ok, json: { email: @signup.email }
+        end
+
+        # POST /api/v1/signups/:token
```
Comment (andresgutgon) [inline engines/account/app/controllers/account/api/v1/signups_controller.rb]: `api/v1/signups/complete/:token`

File: engines/account/config/routes.rb [hunk 3]
```diff
@@ -3,9 +3,9 @@
     namespace :v1 do
       post '/signups', to: 'signups#create'
       get '/signups/:token', to: 'signups#show'
+      post '/signups/:token', to: 'signups#complete'
```
Comment (andresgutgon) [inline engines/account/config/routes.rb]: I find weird to `POST` to the same URL: `/signups/`. 
What is the reason to not use `/signups/complete`?

**Comment by andresgutgon** [comment]
Apart from my question :+1:

File: engines/account/config/routes.rb (hunk 3 above)
Comment (enricostano) [inline engines/account/config/routes.rb]: we are already breaking REST here, so if you prefer `/signups/complete/:token` I can change it

Comment (andresgutgon) [inline engines/account/config/routes.rb]: Yes, more clear for me

File: engines/account/app/models/account/user.rb [hunk 4]
```diff
@@ -5,8 +5,13 @@ class User < ActiveRecord::Base
 
     has_secure_password
 
+    validates :first_name, :last_name, presence: true
```
Comment (enricostano) [inline engines/account/app/models/account/user.rb]: Why?

File: engines/account/app/services/account/signup_service.rb [hunk 5]
```diff
@@ -15,5 +15,30 @@ def create!(email)
...
+        first_name: options[:first_name],
+        last_name: options[:last_name]
+      )
+
+      ::ActiveRecord::Base.transaction do
+        signup.destroy if user.save
+      end
+
+      user
+    end
```
Comment (enricostano) [inline engines/account/app/services/account/signup_service.rb]: We probably need to send a welcome email here

File: engines/account/app/models/account/user.rb (hunk 4 above)
Comment (andresgutgon) [inline engines/account/app/models/account/user.rb]: Why not?

Comment (enricostano) [inline engines/account/app/models/account/user.rb]: Well, we decided that when we created the model and table. If you want to do that we need to change the migration too.

Comment (andresgutgon) [inline engines/account/app/models/account/user.rb]: ok, but you are ok with make first_name and last_name mandatory?

File: engines/account/app/services/account/signup_service.rb (hunk 5 above)
Comment (andresgutgon) [inline engines/account/app/services/account/signup_service.rb]: Yep, do you prefer do it in this PR on in a new one?

Comment (enricostano) [inline engines/account/app/services/account/signup_service.rb]: Don't know, we can merge this to `develop` without and add it later I guess.

File: engines/account/app/models/account/user.rb (hunk 4 above)
Comment (enricostano) [inline engines/account/app/models/account/user.rb]: done

**Pull request merged**